packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3014,6 +3029,34 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     ) {
         this.verifyNotClosed();
         assert(this.connected, 0x259 /* "Container disconnected when trying to submit system message" */);
+
+        if (new TextEncoder().encode(JSON.stringify(contents)).length > 2000) {
+            const originalContentLength = IsoBuffer.from(JSON.stringify(contents)).toString("base64").length;
+
+            const compressionStart = Date.now();
+            const stringifiedContent = JSON.stringify(contents);
+            const contentsAsBuffer = new TextEncoder().encode(stringifiedContent);
+            const compressedContents = compress(contentsAsBuffer);
+            const compressionTime = Date.now() - compressionStart;
+            const compressedContent = IsoBuffer.from(compressedContents).toString("base64");
+
+            this.logger.sendPerformanceEvent({
```

Comment:
> this.logger.sendPerformanceEvent({ [](http://example.com/codeflow?start=11&length=35)

Not sure if you talked to Vlad but do you have an idea on the volume of ops that would be logged in this scenario ? Should we do every 100/1000 or so ?

Reply:
Yeah I don't think this is correct as is. We probably want to do some sort of sampling here if the volume is high.

Reply:
Sampling is a must. I'd tie it somehow to overall op rate. I.e. Ideally amount of telemetry we send should not increase a lot if all ops in a given app are large and compressed.

Reply:
Added some sampling and am logging every 500 compressed ops now. Let me know if that's sufficient or if there's a better strategy for doing this.